src/test/lib/memory/tracking_memory_resource_test.cpp

```diff
@@ -0,0 +1,61 @@
+#include "base_test.hpp"
+#include "memory/tracking_memory_resource.hpp"
+
+namespace opossum {
+
+class TrackingMemoryResourceTest : public BaseTest {
+ protected:
+  TrackingMemoryResource tracking_memory_resource;
+};
+
+TEST_F(TrackingMemoryResourceTest, RecordAllocations) {
+  ASSERT_EQ(tracking_memory_resource.memory_timeseries().size(), 0);
+
+  tracking_memory_resource.allocate(sizeof(int));
```

Comment:
Be explicit with types: Use [u]int(8|16|32|64)_t instead of int, long, uint etc.

Reply:
Furthermore, similar to @nikriek's comment below, this would create a memory leak.